### PR TITLE
Master website snippets nmi

### PR DESCRIPTION
**Pull request by Cocographique**
**Description of the issue/feature this PR addresses:**

https://www.odoo.com/web#id=38878&view_type=form&model=project.task&menu_id=

https://docs.google.com/spreadsheets/d/1ZbX6_QWGwWP7YMxelHEZY9L6Yz1QzcMbAnho7rgdn38/edit?usp=sharing

**Current behavior before PR:**
Ugly snippets w/o options :(

**Desired behavior after PR is merged:**
Nice snippets w/ options :)

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by Cocographique** [comment]
@qsm-odoo :monkey_face:

File: addons/web_editor/static/src/xml/editor.xml [hunk 1]
```diff
@@ -258,6 +258,19 @@
                         </select>
                     </div>
                 </div>
+                <div class="form-group">
+                    <label class="control-label col-sm-3">Style</label>
+                    <div class="col-sm-9">
+                        <select name="link_style_monkey" class="form-control col-sm-3 link-style">
+                            <option value="3d">3D</option>
+                            <option value="flat">Flat</option>
+                            <option value="" selected="selected">Default</option>
```
Comment (qsm-odoo) [inline addons/web_editor/static/src/xml/editor.xml]: Can't go back to this option

File: addons/web_editor/static/src/xml/editor.xml [hunk 2]
```diff
@@ -258,6 +258,19 @@
                         </select>
                     </div>
                 </div>
+                <div class="form-group">
+                    <label class="control-label col-sm-3">Style</label>
+                    <div class="col-sm-9">
+                        <select name="link_style_monkey" class="form-control col-sm-3 link-style">
+                            <option value="3d">3D</option>
+                            <option value="flat">Flat</option>
```
Comment (qsm-odoo) [inline addons/web_editor/static/src/xml/editor.xml]: Are those needed? I tested a bit on Firefox and it is really ugly to me ^^

File: addons/web_editor/static/src/xml/editor.xml [hunk 3]
```diff
@@ -258,6 +258,19 @@
...
+                            <option value="" selected="selected">Default</option>
+                            <option value="outline">Outline</option>
+                            <option value="rounded">Rounded</option>
+                            <option value="rounded_outline">Outline &amp; Rounded</option>
```
Comment (qsm-odoo) [inline addons/web_editor/static/src/xml/editor.xml]: Definitely something we need to do after going to bootstrap 4, as it is standard in there

File: addons/web_editor/static/src/js/widgets/widgets.js [hunk 4]
```diff
@@ -1459,8 +1459,10 @@ var LinkDialog = Dialog.extend({
 
         var style = this.$('input[name="link_style_color"]:checked').val() || '';
         var size = this.$('select[name="link_style_size"]').val() || '';
+        var monkey = this.$('select[name="link_style_monkey"]').val() || '';
```
Comment (qsm-odoo) [inline addons/web_editor/static/src/js/widgets/widgets.js]: Seriously? :p

File: addons/website/static/src/less/website.less [hunk 5]
```diff
@@ -6,6 +6,22 @@ font[style*='background'], font[class*='bg-'] {
...
+// Padding
+
+.s_padding_none { padding: 0; }
+.s_padding_default { padding: 16px; }
+
+.s_padding_xs { padding: 8px;  }
+.s_padding_sm { padding: 16px; }
+.s_padding_md { padding: 32px; }
+.s_padding_lg { padding: 48px; }
+.s_padding_xl { padding: 64px; }
```
Comment (qsm-odoo) [inline addons/website/static/src/less/website.less]: Seems like something for bootstrap 4

File: addons/website/static/src/less/website.less [hunk 6]
```diff
@@ -59,23 +75,175 @@ font[style*='background'], font[class*='bg-'] {
...
-        padding: @s_text_block_image_fw-padding;
-        > .row > div {
-            padding: 30px;
+// "Title" snippet
+
+@s_title_transparent-opacity: .5;
+@s_title_thin-font-weight: 300;
+
+.s_title {
+    [class*="s_title_"] {
```
Comment (qsm-odoo) [inline addons/website/static/src/less/website.less]: Do not use selector like those ;) Use `.s_title_boxed, .s_title_lines, ...`

File: addons/website/static/src/less/website.less [hunk 7]
```diff
@@ -59,23 +75,175 @@ font[style*='background'], font[class*='bg-'] {
...
+            &.bg-primary { box-shadow: 0 0 0 5px currentColor, 0 0 0 10px @brand-primary; }
+            &.bg-info { box-shadow: 0 0 0 5px currentColor, 0 0 0 10px @brand-info; }
+            &.bg-sucess { box-shadow: 0 0 0 5px currentColor, 0 0 0 10px @brand-success; }
+            &.bg-warning { box-shadow: 0 0 0 5px currentColor, 0 0 0 10px @brand-warning; }
+            &.bg-danger { box-shadow: 0 0 0 5px currentColor, 0 0 0 10px @brand-danger; }
+            &.bg-gray-lighter { box-shadow: 0 0 0 5px currentColor, 0 0 0 10px @gray-lighter; }
+            &.bg-gray-light { box-shadow: 0 0 0 5px currentColor, 0 0 0 10px @gray-light; }
+            &.bg-gray { box-shadow: 0 0 0 5px currentColor, 0 0 0 10px @gray; }
+            &.bg-gray-dark { box-shadow: 0 0 0 5px currentColor, 0 0 0 10px @gray-dark; }
+            &.bg-gray-darker { box-shadow: 0 0 0 5px currentColor, 0 0 0 10px @gray-darker; }
```
Comment (qsm-odoo) [inline addons/website/static/src/less/website.less]: Wow, what is this? :p

File: addons/website/static/src/less/website.less [hunk 8]
```diff
@@ -500,6 +664,256 @@ div.carousel {
...
+        }
+    }
+}
+
+.o-create-panel-variant-classes(@name, @color, @complementary: contrast(@color, @text-color, #FFF, 60%)) {
+    .panel-@{name} {
+        .panel-variant(@color; @complementary; @color; @color);
+    }
+}
+
```
Comment (qsm-odoo) [inline addons/website/static/src/less/website.less]: Definitely something we will not do since those panel are gone in bootstrap 4

**Pull request closed without merging**